hack/e2e/eventing/peerauthentications/peerauthentications_test.go

```diff
@@ -0,0 +1,133 @@
+//go:build e2e
+// +build e2e
+
+package peerauthentications
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	istiosecv1beta1 "istio.io/api/security/v1beta1"
+	istio "istio.io/client-go/pkg/apis/security/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common"
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common/fixtures"
+	"github.com/kyma-project/eventing-manager/hack/e2e/common/testenvironment"
+	"github.com/stretchr/testify/require"
+)
+
+var testEnvironment *testenvironment.TestEnvironment
+
+// TestMain runs before all the other test functions.
+func TestMain(m *testing.M) {
+	testEnvironment = testenvironment.NewTestEnvironment()
+
+	// Wait for eventing-manager deployment to get ready.
+	if err := testEnvironment.WaitForDeploymentReady(ManagerDeploymentName, NamespaceName, ""); err != nil {
+		testEnvironment.Logger.Fatal(err.Error())
+	}
+
+	// Run the tests and exit.
+	code := m.Run()
+	os.Exit(code)
+}
+
+// Test_EventPublisherProxyMetricsPeerAuthentication checks if the Istio PeerAuthentication for the metrics endpoint of
+// Event-Publisher-Proxy was created.
+func Test_EventPublisherProxyMetricsPeerAuthentication(t *testing.T) {
```

Comment:
there is no way to figure out what this function tests without reading the comment

Reply:
renamed it to make it more clearer but not sure.

Reply:
better than before 😄